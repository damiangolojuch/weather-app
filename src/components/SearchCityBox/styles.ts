import styled from "styled-components";
import { Input } from "antd";

export const SearchStyled = styled(Input.Search)``;
