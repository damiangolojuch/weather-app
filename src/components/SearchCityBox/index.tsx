import * as S from "./styles";
import { ChangeEvent, MouseEvent, FC } from "react";
import { Button, Form } from "antd";

export interface SearchCityBoxProps {
  className?: string;
  currentCityName: string;
  onSearch(
    text: string,
    event?: ChangeEvent<any> | MouseEvent | KeyboardEvent
  ): void;
}

const SearchCityBox: FC<SearchCityBoxProps> = ({
  className,
  currentCityName,
  onSearch,
}) => (
  <Form action={"/"} method="GET">
    <S.SearchStyled
      name="city"
      className={className}
      placeholder="Wpisz miejscowość"
      enterButton={
        <Button type="primary" htmlType="submit">
          Szukaj
        </Button>
      }
      size="large"
      defaultValue={currentCityName}
      onSearch={onSearch}
    />
  </Form>
);

export default SearchCityBox;
