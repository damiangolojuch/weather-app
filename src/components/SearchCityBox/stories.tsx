import { ComponentMeta, ComponentStory } from "@storybook/react";
import SearchCityBox from ".";

export default {
  title: "SearchCityBox",
  component: SearchCityBox,
  args: {
    description: "Antd Input.Search adjusted to city search",
  },
  argTypes: {
    onSearch: {
      action: "onSearch",
    },
  },
} as ComponentMeta<typeof SearchCityBox>;

const DefaultTemplate: ComponentStory<typeof SearchCityBox> = (args) => (
  <SearchCityBox {...args} />
);

export const Default = DefaultTemplate.bind({});

Default.args = {
  currentCityName: "",
};
