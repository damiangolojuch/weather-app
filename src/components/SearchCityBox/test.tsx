import SearchCityBox, { SearchCityBoxProps } from ".";
import { render } from "@testing-library/react";

const setup = (customProps: Partial<SearchCityBoxProps> = {}) => {
  const defaultProps: SearchCityBoxProps = {
    currentCityName: "",
    onSearch: jest.fn(),
    ...customProps,
  };

  return render(<SearchCityBox {...defaultProps} />);
};

describe("<SearchCityBox />", () => {
  it(`should render antd Input.Search`, () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
