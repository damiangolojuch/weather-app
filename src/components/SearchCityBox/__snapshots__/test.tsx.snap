// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchCityBox /> should render antd Input.Search 1`] = `
<div>
  <form
    action="/"
    class="ant-form ant-form-horizontal"
    method="GET"
  >
    <span
      class="ant-input-group-wrapper ant-input-search ant-input-search-large ant-input-search-with-button sc-bczRLJ ant-input-group-wrapper-lg"
    >
      <span
        class="ant-input-wrapper ant-input-group"
      >
        <input
          class="ant-input ant-input-lg"
          name="city"
          placeholder="Wpisz miejscowość"
          type="text"
          value=""
        />
        <span
          class="ant-input-group-addon"
        >
          <button
            class="ant-btn ant-btn-primary ant-btn-lg ant-input-search-button"
            type="submit"
          >
            <span>
              Szukaj
            </span>
          </button>
        </span>
      </span>
    </span>
  </form>
</div>
`;
